arwen/mock/epochNotifierStub.go

```diff
@@ -0,0 +1,28 @@
+package mock
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// EpochNotifierStub -
+type EpochNotifierStub struct {
+	CurrentEpochCalled          func() uint32
+	RegisterNotifyHandlerCalled func(handler vmcommon.EpochSubscriberHandler)
+}
+
+// RegisterNotifyHandler -
+func (ens *EpochNotifierStub) RegisterNotifyHandler(handler vmcommon.EpochSubscriberHandler) {
+	if ens.RegisterNotifyHandlerCalled != nil {
+		ens.RegisterNotifyHandlerCalled(handler)
+	} else {
+		if !check.IfNil(handler) {
+			handler.EpochConfirmed(0, 0)
+		}
+	}
+}
+
+// IsInterfaceNil -
+func (ens *EpochNotifierStub) IsInterfaceNil() bool {
+	return ens == nil
+}
```

Comment:
Won't this give a nil pointer exception if ens is nil? (i.e. `ens.IsInterfaceNil()`)

Reply:
we are using check.IfNil function which is doing if object == nil || object.IsInterfaceNil return true